espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/adapter/espresso/EspressoW3CActionAdapter.java

```diff
@@ -480,8 +483,12 @@ public void waitForUiThread() {
         uiController.loopMainThreadUntilIdle();
     }
 
-    public void sleep(long duration) throws AppiumException {
-        uiController.loopMainThreadForAtLeast(duration);
+    public void sleep(float duration) {
+        long roundedDuration = Math.round(duration);
+        if (duration != roundedDuration) {
```

Comment:
`float != int` should always be true. One must cast float to an integer type first in order to compare it for equality

Reply:
I just did this test:

```java

        long numOne = 10;
        float numTwo = 10.0F;
        boolean comp = numOne == numTwo;
```

And `comp` evaluates to true.

I think that when comparisons are made, `long` is cast to `float`.